backend/src/main.rs

```diff
@@ -247,26 +248,98 @@ fn submit_task(
 }
 
 #[query]
-fn get_task(_id: TaskId) -> Task {
-    Task {
+fn get_task(id: TaskId) -> Task {
+    let caller = caller();
+    get_task_impl(caller, id)
+}
+
+fn get_task_impl(caller: Principal, id: TaskId) -> Task {
+    let mut task: Task = Task {
         submitter: Principal::anonymous(),
         task_type: TaskType::TranslateText,
         payload: ByteBuf::from(vec![]),
         deadline: 0,
         reward: 0,
         answers: vec![],
         status: TaskStatus::Open,
-    }
+    };
+
+    STATE.with(|s| {
+        let ledger = s.ledger.borrow();
+        if !ledger.contains_key(&caller) {
+            ic_cdk::trap(&format!("{} has not been registered yet.", caller));
+        }
+        let task_map = s.tasks.borrow();
+        if !task_map.contains_key(&id) {
+            ic_cdk::trap(&format!("Requested task id {} cannot be found", id));
+        }
+        let answers_map = s.answers.borrow();
+        if let Some(task_internal_ref) = task_map.get(&id) {
+            let task_internal : TaskInternal = (*task_internal_ref).clone();
+            let mut answers:Vec<Answer> = Vec::new();
+            for ans_id in task_internal.answers {
+                if !answers_map.contains_key(&ans_id) {
+                    ic_cdk::trap(&format!("Inconsistent state. AnswerId {} cannot be found", ans_id));
+                }
+                if let Some(ans_ref) = answers_map.get(&ans_id){
+                    answers.push(ans_ref.clone());
+                }
+            }
+            task = Task {
+                submitter: task_internal.submitter,
+                task_type: task_internal.task_type,
+                payload: task_internal.payload,
+                deadline: task_internal.deadline,
+                reward: task_internal.reward,
+                answers,
```

Comment:
I am unclear whether the task contains the answers directly or whether it just contains the answer_ids. 
It it the case that the taskInternal only contains the answer_ids but there is also a struct Task which contains the answers directly and we just send this back here?
If this is the case, do we need the distinction at all or couldn't we just use the struct which directly has the answers in the tasks (maybe even without answer_ids)?